sstables/sstable_directory.cc

```diff
@@ -109,19 +109,24 @@ void sstable_directory::validate(sstables::shared_sstable sst, process_flags fla
     }
 }
 
-future<>
-sstable_directory::process_descriptor(sstables::entry_descriptor desc, process_flags flags) {
-    if (desc.version > _max_version_seen) {
-        _max_version_seen = desc.version;
-    }
-
+future<sstables::shared_sstable> sstable_directory::load_sstable(sstables::entry_descriptor desc, process_flags flags) const {
     auto sst = _manager.make_sstable(_schema, _sstable_dir.native(), desc.generation, desc.version, desc.format, gc_clock::now(), _error_handler_gen);
     co_await sst->load(_io_priority);
```

Comment:
Just a random observation that the pattern
```
sst = manager.make_sstable();
co_await sst->load()
return sst
```
is actively used in tests. Maybe it will deserve a helper sugar on manager

Reply:
That's true. To be honest, I really hate the fact that sstable became a sink for all operations around SSTable. In my opinion there should be three classes: sstable, sstable_loader, sstable_writer. it's unclear if a sstable is loaded or not. in my opinion, sstable being loaded should be an invariant. sstable_loader loads components for a given sstable and constructs a sstable on completion. Likewise, sstable_writer becomes the sole responsible for writing new sstables and provide a sstable on completion. today, a sstable can be in unloaded, loaded, 'partially written' states. what a mess, no?

Reply:
does this imply that, with `sstable_loader` , `read_toc()` and `read_scylla_metadata()` will be methods of this class ?

Reply:
yes, but that's for the long term. I think the first good step is to split sstable and sstable writer related methods completely. Please note sstable_writer already exists, but we haven't completed the split.